guide/src/module.md

```diff
@@ -48,53 +51,133 @@ To import the module, either:
 
 ## Documentation
 
-The [Rust doc comments](https://doc.rust-lang.org/stable/book/first-edition/comments.html) of the module
+The [Rust doc comments](https://doc.rust-lang.org/stable/book/ch03-04-comments.html) of the module
 initialization function will be applied automatically as the Python docstring of your module.
 
+For example, building off of the above code, this will print `This module is implemented in Rust.`:
+
 ```python
-import rust2py
+import my_extension
 
-print(rust2py.__doc__)
+print(my_extension.__doc__)
 ```
 
-Which means that the above Python code will print `This module is implemented in Rust.`.
+## Organizing your module registration code
 
-## Modules as objects
+For most projects, it's adequate to centralize all your FFI code into a single Rust module.
 
-In Python, modules are first class objects. This means that you can store them as values or add them to
-dicts or other modules:
+However, for larger projects, it can be helpful to split your Rust code into several Rust modules to keep your code 
+readable. Unfortunately, though, some of the macros like `wrap_pyfunction!` 
+[do not yet work](https://github.com/PyO3/pyo3/issues/1709) when used on code defined in other 
+modules. One way to work around this is to pass references to the `PyModule` so that each module registers its own 
+FFI code. For example:
 
 ```rust
+// src/lib.rs
+use pyo3::prelude::*;
+
+#[pymodule]
+fn my_extension(py: Python, m: &PyModule) -> PyResult<()> {
+    dirutil::register(py, m)?;
+    osutil::register(py, m)?;
+    Ok(())
+}
+
+// src/dirutil.rs
+# mod dirutil {
+use pyo3::prelude::*;
+
+pub(crate) fn register(py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<SomeClass>()?;
+    Ok(())
+}
+
+#[pyclass]
+struct SomeClass {
+    x: usize,
+}
+# }
+
+// src/osutil.rs
+# mod osutil {
 use pyo3::prelude::*;
-use pyo3::wrap_pymodule;
-use pyo3::types::IntoPyDict;
+
+pub(crate) fn register(py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(determine_current_os, m)?)?;
+    Ok(())
+}
 
 #[pyfunction]
-fn subfunction() -> String {
-    "Subfunction".to_string()
+fn determine_current_os() -> String {
+    "linux".to_owned()
 }
+# }
+```
+
+Another workaround for splitting FFI code across multiple modules is to add `use module::*`, like this: 
```

Comment:
It might be worth adding a quick note about #1709, both so users can understand why the glob is suggested (which is usually seen as an antipattern) and as a reminder for us to improve it when we implement a better solution.